### PR TITLE
Feature/93 unit tests backend connector services

### PR DESCRIPTION
**Pull request by gregdeane**
#93

### CODE REVIEW

File: src/app/common/backend-connector/backend-connector.service.spec.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  function instantiate() {
+    backendConnectorService = new BackendConnectorService($window, mock.environmentsService);
+  }
+
+  function spyOnEnvironmentsService(method, value) {
+    mock.environmentsService[method].and.returnValue(value);
+  }
+
+  function teardown() {
+    $window['jasmine'] = $windowJasmineCopy;
```
Comment (dami-gg) [inline src/app/common/backend-connector/backend-connector.service.spec.js]: Just out of curiosity, what's this for?

File: src/app/common/backend-connector/backend-connector.service.spec.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  });
+
+  it('should check if override is possible for LOCAL', () => {
+    spyOnEnvironmentsService('getCurrentEnvironment', { NAME: mock.environment.LOCAL });
+    let isOverridePossible = backendConnectorService.isOverridePossible();
+
+    expect(isOverridePossible).toBe(true);
+  });
+
+  it('should check if override is possible for INTEGRATION', () => {
```
Comment (dami-gg) [inline src/app/common/backend-connector/backend-connector.service.spec.js]: Would you mind adding one to verify that override is NOT possible in production? That's the crucial one I believe Thanks!

Comment (gregdeane) [inline src/app/common/backend-connector/backend-connector.service.spec.js]: Sure!

File: src/app/common/backend-connector/backend-connector.service.spec.js (hunk 1 above)
Comment (gregdeane) [inline src/app/common/backend-connector/backend-connector.service.spec.js]: It makes it possible to test the `getSpecificEnvironment()` branch of `getEnvironment()`.

File: src/app/common/backend-connector/backend-connector.service.spec.js (hunk 2 above)
Comment (gregdeane) [inline src/app/common/backend-connector/backend-connector.service.spec.js]: Done

File: src/app/common/backend-connector/backend-connector.service.spec.js (hunk 1 above)
Comment (gregdeane) [inline src/app/common/backend-connector/backend-connector.service.spec.js]: If it's not clear what I mean, let me know.

Comment (dami-gg) [inline src/app/common/backend-connector/backend-connector.service.spec.js]: oh yes, I think I've got it now Thanks ;)

**Comment by dami-gg** [comment]
:+1:

**Comment by gregdeane** [comment]
👍

**Pull request merged**